File: src/test/java/integration/BaseIntegrationTest.java
package integration;

import com.codeborne.selenide.Configuration;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;

import static com.codeborne.selenide.Selenide.closeWebDriver;
import static com.codeborne.selenide.Selenide.open;

public abstract class BaseIntegrationTest {
    @BeforeAll
    static void setUp() {
         Configuration.timeout = 5000;
    }

    @BeforeEach
    void openURl() {
        open("");
    }

    @AfterEach
    void closeBrowser() {
        closeWebDriver();
    }
}

File: src/test/kotlin/com/github/qky666/selenidepom/test/kotlin/BaseIntegrationTest.kt
package com.github.qky666.selenidepom.test.kotlin

import com.codeborne.selenide.Configuration
import com.codeborne.selenide.Selenide
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.AfterEach

abstract class BaseIntegrationTest {
    companion object {
        @BeforeAll
        @JvmStatic
        fun setUp() {
            Configuration.timeout = 5000
        }
    }

    @BeforeEach
    fun openURl() {
        Selenide.open("")
    }

    @AfterEach
    fun closeBrowser() {
        Selenide.closeWebDriver()
    }
}
